state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -698,6 +675,94 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Request a swap deposit address.
+		///
+		/// ## Events
+		///
+		/// - [SwapDepositAddressReady](Event::SwapDepositAddressReady)
+		#[pallet::call_index(10)]
+		#[pallet::weight(T::WeightInfo::request_swap_deposit_address())]
+		pub fn request_swap_deposit_address_v2(
+			origin: OriginFor<T>,
+			source_asset: Asset,
+			destination_asset: Asset,
+			destination_address: EncodedAddress,
+			broker_commission: BrokerFees<T::AccountId>,
+			channel_metadata: Option<CcmChannelMetadata>,
+			boost_fee: BasisPoints,
+		) -> DispatchResult {
+			ensure!(T::SafeMode::get().deposits_enabled, Error::<T>::DepositsDisabled);
+			let broker = T::AccountRoleRegistry::ensure_broker(origin)?;
+			let (beneficiaries, total_bps): (
+				BoundedVec<Beneficiary<T::AccountId>, ConstU32<MAX_BENEFICIARIES>>,
+				BasisPoints,
+			) = match broker_commission {
+				BrokerFees::Single(broker_bps) =>
+					if broker_bps > 0 {
+						(
+							vec![Beneficiary { account: broker.clone(), bps: broker_bps }]
+								.try_into()
+								.unwrap(),
+							broker_bps,
+						)
+					} else {
+						(vec![].try_into().unwrap(), broker_bps)
+					},
+				BrokerFees::Multiple(ref fees_bps) => {
+					for Beneficiary { account, bps: _ } in fees_bps.clone() {
+						ensure!(
+							T::AccountRoleRegistry::has_account_role(&account, AccountRole::Broker),
+							Error::<T>::BeneficiaryAccountIsNotABroker
+						);
+					}
+					(
+						(*fees_bps)
+							.clone()
+							.into_iter()
+							.filter(|Beneficiary { account: _, bps }| *bps > 0)
+							.collect::<Vec<_>>().try_into().expect("The provided beneficiaries vec cannot have more than MAX_BENEFICIARIES elements"),
```

Comment:
it would be more useful to the user (and better for us because we have to spend less time answering this question) to include the actual maximum count in the error message. As it stands, this is just a constant in code, so the user can't look up the correct number either without diving into the backend code afaik.

Reply:
This isn't the error message - Think of it as more like a code comment about why we never expect to hit this case.
If anyone tries to send too many beneficiaries, it will cause a decoding error and should bubble up to the surface in the broker api (tbc).

Reply:
Yeah can confirm, just tried to submit through the broker api and directly using the new extrinsic and it fails in both cases during decoding for the broker and during verification for the extrinsic.

Reply:
This isn't an error message that anyone will ever see - Think of it as more like a code comment about why we never expect to hit this case.

If anyone tries to send too many beneficiaries, it will cause a decoding error and should bubble up to the surface in the broker api (i'll check what this actually returns in practice).

Reply:
And if you try to send an extrinsic with too many values, it will return a decoding error too. If you're using a substrate framework, it should protect against this. 

If you're rolling your own, you should still get a sensible error message about failing to decode the BoundedVec.

Reply:
`data did not match any variant of untagged enum BrokerFees` can be improved in the broker API, I agree that if you are directly submitting the extrinsic yourself, you should be prepared to debug the error messages (even though in practice it means they will ping us)

Reply:
Yeah I also think that the use case of splitting your fees between 5 accounts is remote enough that we can accept the risk that the error message might not be perfect.

BrokerFees is another matter - can definitely be improved.

Reply:
See: [cac5f7e](https://github.com/chainflip-io/chainflip-backend/pull/4777/commits/cac5f7e4f3ad02601989e03f40cad9dfec26e175)